### DISCUSSION TITLE
Licensing - stork, SVG icons, etc.

### DISCUSSION THREAD

**Discussion by kukimik** [srid/emanote#373] [Q&A]
I'm not a lawyer, but shouldn't a copy of the Apache License 2.0 file be provided together with the stork files? https://github.com/jameslittle230/stork/blob/8550f1fe13c334f16026d53886d3e17ae0193925/license.txt#L94 makes me think so.

And maybe there's also a similar issue with the SVG icons (and perhaps some other assets that don't come to mind at the moment).

The current state of the project may suggest that it is licensed under https://github.com/EmaApps/emanote/blob/master/LICENSE as a whole, which - as far as I understand - it is not.

**Comment by srid** [comment]
What do you think should be done to address this?

I chose AGPL to avoid the situation of people extending Emanote, creating a proprietary web application, and not contributing changes back. Otherwise, any liberal open-source/free software licensing would do, as far as I'm concerned.

**Reply by kukimik** [reply]
I think the problem has 2 levels:
* external code that is included in the repo;
* external code that is included in the generated websites.
  * how to treat including JS/CSS via CDN? this is not clear to me

To address the first one, I think we could clearly state (in the README? in some other files? in a separate LICENSES.TXT file?) which assets are licensed under which license. And include the license texts when appropriate. @toastal 's suggestion regarding SVG metadata is also worth considering.

Regarding the second one, perhaps an extra page could be added to the generated websites (say, under a :balance_scale: icon) that would lead to a page that looks like https://elementor.com/licenses/ or https://opensource.stackexchange.com/a/11865 

That's what I feel should suffice in most cases (Apache 2.0, MIT,...). But don't take it as a legal advice, I know next to nothing about licensing issues. It's just the result of my intuition plus a few minutes of googling.

**Reply by toastal** [reply]
GNU also has a bit of a spec for laying out JavaScript licenses https://www.gnu.org/licenses/javascript-labels.html

I like the explicit call out to the projects and giving the recognition. I still think it's a good idea to at minimum keep a SPDX license identifier with the source if possible (be that a comment, adding EXIF data to images, etc.)

**Comment by srid** [comment]
Also, I'd really like to change the Emanote logo to something nicer.

**Reply by kukimik** [reply]
Where does the logo come from? (Why is it relevant to this discussion?)

How should the new logo look like? 

This should probably be moved to a separate issue/discussion.

**Reply by srid** [reply]
> This should probably be moved to a separate issue/discussion.

https://github.com/EmaApps/emanote/discussions/379

> How should the new logo look like?

No idea, TBH. Something that connotes the act of emanating a complex structure out of a flat list of Markdown files?

**Comment by toastal** [comment]
For SVGs, there are namespaces to help out with licensing properly.

```
  <metadata id="license"
            xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
            xmlns:dc="http://purl.org/dc/elements/1.1/"
            xmlns:cc="http://creativecommons.org/ns#">
```

NOTE: when optimizing SVGs should 99% of the time choose `scour` over `svgo` as `svgo` will strip out the licensing info as "editor data" and by default because their creator thinks smallest size is always the optimal goal instead of respecting licenses (or even rendering correctly... look at how many bugs are in `svgr` for React because of `svgo`).

**Comment by srid** [comment]
Licenses are included in footnote of README: https://github.com/EmaApps/emanote

If I missed anything, let me know.
